src/main/webapp/app/shared/metis/metis.service.ts

```diff
@@ -597,6 +596,16 @@ export class MetisService implements OnDestroy {
             case MetisPostAction.UPDATE:
                 const indexToUpdate = this.cachedPosts.findIndex((post) => post.id === postDTO.post.id);
                 if (indexToUpdate > -1) {
+                    // WebSocket does not currently update the author and authorRole of posts correctly, so this is implemented as a workaround
+                    postDTO.post.author = this.cachedPosts[indexToUpdate].author;
```

Comment:
Are you sure this is necessary? In my testing, the author attribute returned by the websocket was always there. Just authorRole was missing

Reply:
Yes you are right, only authorRole is missing. I removed the unnecessary line